ui/src/mockState.tsx

```diff
@@ -50,8 +50,10 @@ export const localState: LocalStorage = {
 const history = createMemoryHistory({entries: ['/']})
 
 export function renderWithRedux(ui, initialState = s => s) {
+  clearStore()
   const seedStore = configureStore(localState, history)
   const seedState = seedStore.getState()
+  clearStore()
```

Comment:
to share the store around the app without recreating stores, i had to make it a singleton, which makes sense for a single page app. maybe a slight interface change to allow for multiple stores to be created will be required in the future, but the desired outcome is to provide the developer the ability to say "give me the store" without explicitly plumbing it everywhere. There be some abstraction dragons here, but i like to think we're using the store as a singleton anyways, just allowing `connect` to be that interface that obfuscates the plumbing for us, so it's not THAT bad. the clearStore works like how we use the `/debug/flush` endpoint for e2e tests, so that we can inject fresh data into the singleton on every test. This specific stuff looks weird because we're using a store to load a store, but overwrite specific things, instead of injecting a default store with overrides, which i can imagine is to allow for greater flexibility in _how_ that default store gets defined / created and more easily mirror actual application state than keeping a derived state in sync with an explicit one.